cpp/src/io/json/write_json.cu

```diff
@@ -254,38 +182,200 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
                            strings_columns.end(),
                            [](auto const& c) { return c.type().id() == type_id::STRING; }),
                "All columns must be of type string");
+  auto constexpr str_views_per_column = 3;  // (for each "column_name:", "value",  "separator")
+  auto const num_str_views_per_row    = strings_columns.num_columns() * str_views_per_column + 1;
+  // Eg. { col1: value , col2: value , col3: value } = 1+ 3+3+(3-1) +1 = 10
+
+  auto tbl_device_view = cudf::table_device_view::create(strings_columns, stream);
+  auto d_column_names  = column_device_view::create(column_names, stream);
+
+  // (num_columns*3+1)*num_rows size (very high!) IDEA: chunk it here? but maximize parallelism?
+  auto const string_views_total = num_str_views_per_row * strings_columns.num_rows();
+  auto const total_rows         = strings_columns.num_rows() * strings_columns.num_columns();
+  using str_pair                = thrust::pair<const char*, size_type>;
+  rmm::device_uvector<str_pair> d_strviews(string_views_total, stream);
+
+  // scatter row_prefix, row_suffix, column_name:, value, value_separator as string_views
+  thrust::for_each(
+    rmm::exec_policy(stream),
+    thrust::make_counting_iterator<size_type>(0),
+    thrust::make_counting_iterator<size_type>(total_rows),
+    [tbl       = *tbl_device_view,
```

Comment:
This is a large lambda and capture group. Could it be made into a functor or do you feel it is better inline?

Reply:
moved to lambda.